src/api/app/helpers/webui/webui_helper.rb

```diff
@@ -411,4 +411,17 @@ def word_break(string, length)
     # adds a <wbr> tag after an amount of given characters
     safe_join(string.scan(/.{1,#{length}}/), "<wbr>".html_safe)
   end
+
+  def toggle_sliced_text(text, slice_length = 50, id = "toggle_sliced_text_#{Time.now.to_f.to_s.delete('.')}")
+    return text if text.length < slice_length
+    short = content_tag(:span, 'data-toggle-id' => id) do
+      content_tag(:span, text.slice(0, slice_length) + ' ') +
+      link_to('[+]', '#', onclick: "$(\"[data-toggle-id='#{id}']\").toggle();".html_safe)
```

Comment:
I really don't like the variable here as if id is some js we have a XSS injection!

```
'$(\"[data-toggle-id='.html_safe + id + '").toggle();'.html_safe
```
would be the safest option!